### ISSUE TITLE
Any way to disable automatic restore on each recompile?

### ISSUE THREAD

**Issue by lehtiniemi** [GlitchEnzo/NuGetForUnity#407]
## Description
Is there any way to disable restoring packages on every recompile (and maybe just do it on Unity start)? With just a couple NuGet-packages installed, Unity restores them on each recompile and this is huge amount of recompile time per day.

- **NuGet Package:** [What is the name and version of the NuGet package you are installing?]
![image](https://user-images.githubusercontent.com/13351978/124436077-4e0c9f80-dd7e-11eb-9487-e9a0d53489ad.png)

- **NuGetForUnity Version:** [What version of NuGetForUnity are you running?]
- 3.0.2
- **Unity Version:** [What version of Unity are you using?]
- 2021.2.0a16
- **Operating System:** [What OS are you on?]
- OSX 10.15.7

**Comment by jwittner** [comment]
Hmmm, it should be detecting that they're already installed and skipping any installation and avoiding any recompiling by doing so. I wonder if this is only on OSX?

**Comment by NoTuxNoBux** [comment]
> I wonder if this is only on OSX?

Negative, I can confirm I'm also experiencing it on Windows with Unity 2020.3.11f1 :slightly_smiling_face: .

I don't recall it happening with Unity 2019.4.27, but Unity 2019 didn't show a progress bar with what it was doing on recompile, so it may well have been.

**Comment by AlexanderLuck** [comment]
We just got the package a week ago and have the same issue on Windows. 
Just 2 packages adds 4-5 seconds every time we compile, which makes using this package hard to justify.
Is this being worked on?

**Comment by jwittner** [comment]
That does seem excessive for determining that the packages required are already installed. No one is looking into this at the moment that I know of.

Would love:
* an investigation into why it's slow.
* a personal preference in Editor -> Preferences to control the automatic restore.

PRs are welcome! =)

**Comment by Shii2** [comment]
As a workaround for now I added variable:
`private const string k_ProjectOpened = "ProjectOpened";` to the NugetHelper.cs
and guarded Restore with this code:
```
if (!SessionState.GetBool(k_ProjectOpened, false))
{
    SessionState.SetBool(k_ProjectOpened, true);
    Restore();
}
```
After this change Nuget only calls Restore one time after loading the project and not calling after changing any scripts in the project.

**Comment by NoTuxNoBux** [comment]
@Shii2 Thanks for the tip. I formalized your changes in PR #424. I don't see any immediate loss of functionality by simply doing it like this and, well, sweet, sweet, reduced compilation times in my projects now :smile:.

**Comment by NoTuxNoBux** [comment]
I think this can be closed, since the fix was published in 3.0.4.
